### PR TITLE
feat(EMS-4034): declarations - modern slavery - data model/creation

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR sets up the initial data model and updates application creation for the new "declarations - modern slavery" questions.

## Resolution :heavy_check_mark:
- Update `DECLARATIONS` constants.
- Create new helper functions:
  - `createADeclarationModernSlavery`
  - `createADeclarationModernSlaveryVersion`
- Update `createADeclaration` helper function to call `createADeclarationModernSlavery`.
- Create new `DeclarationModernSlavery` and `DeclarationModernSlaveryVersion` DB tables.
- Create new test helper function, `declarationsModernSlavery.create`.
- Update types.
- Update the UI's "get application" GQL query.

## Miscellaneous :heavy_plus_sign:
- Fix some typos.
- Simplify some types.

### CODE REVIEW

File: src/api/constants/declarations/modern-slavery-versions/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import DECLARATION_VERSIONS from '.';
+
+describe('api/constants/declarations/modern-slavery-versions', () => {
+  it('should return an array of modern slavery decalration versions', () => {
```
Comment (Zainzzkk) [inline src/api/constants/declarations/modern-slavery-versions/index.test.ts:4]: typo

File: src/api/constants/declarations/versions/index.ts [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+  CONFIRMATION_AND_ACKNOWLEDGEMENTS: '1',
+  HOW_YOUR_DATA_WILL_BE_USED: '1',
+};
+
+const VERSION_2: ApplicationDeclarationVersions = {
+  ANTI_BRIBERY: '2',
+  ANTI_BRIBERY_CODE_OF_CONDUCT: '2',
+  ANTI_BRIBERY_EXPORTING_WITH_CODE_OF_CONDUCT: '1',
+  CONFIDENTIALITY: '1',
+  CONFIRMATION_AND_ACKNOWLEDGEMENTS: '1',
```
Comment (Zainzzkk) [inline src/api/constants/declarations/versions/index.ts:17]: are the new ones not included in this file?

Comment (ttbarnes) [reply src/api/constants/declarations/versions/index.ts:17]: No, these are the top level declaration versions. The new child declaration versions for modern slavery are in `DECLARATION_MODERN_SLAVERY_VERSIONS`

File: src/api/helpers/create-a-declaration-modern-slavery-version/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  test('it should return declaration version fields with the latest versions', async () => {
+    const result = await createADeclarationModernSlaveryVersion(context, declarationModernSlavery.id);
+
+    expect(result.willAdhereToAllRequirements).toEqual(WILL_ADHERE_TO_ALL_REQUIREMENTS);
+    expect(result.hasNoOffensesOrInvestigations).toEqual(HAS_NO_OFFENSES_OR_INVESTIGATIONS);
+    expect(result.isNotAwareOfExistingSlavery).toEqual(IS_NOT_AWARE_OF_EXISTING_SLAVERY);
+  });
+
+  describe('when an invalid declaration modern slavery ID is passed', () => {
+    test('it should throw an error', async () => {
```
Comment (Zainzzkk) [inline src/api/helpers/create-a-declaration-modern-slavery-version/index.test.ts:51]: could we use .toThrow?

File: src/api/helpers/create-a-declaration-modern-slavery/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  describe('when an invalid declaration ID is passed', () => {
+    test('it should throw an error', async () => {
+      await expect(createADeclarationModernSlavery(context, mockInvalidId)).rejects.toThrow('Creating an application declaration modern slavery');
+    });
+  });
+
+  describe('when creation is not successful', () => {
+    test('it should throw an error', async () => {
+      await expect(createADeclarationModernSlavery({}, application.id)).rejects.toThrow('Creating an application declaration modern slavery');
+      try {
```
Comment (Zainzzkk) [inline src/api/helpers/create-a-declaration-modern-slavery/index.test.ts:84]: same as above

File: src/api/helpers/create-a-declaration-modern-slavery/index.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  try {
+    const declarationModernSlavery = await context.db.DeclarationModernSlavery.createOne({
+      data: {
+        declaration: {
+          connect: { id: declarationId },
+        },
+      },
+    });
+
+    const declarationModernSlaveryVersion = await createADeclarationModernSlaveryVersion(context, declarationModernSlavery.id);
```
Comment (Zainzzkk) [inline src/api/helpers/create-a-declaration-modern-slavery/index.ts:23]: do we need to optional chain .id incase it doesn't exist?

File: src/api/helpers/create-a-declaration-modern-slavery-version/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    declarationModernSlavery = (await declarationsModernSlavery.create(context)) as ApplicationDeclarationModernSlavery;
+  });
+
+  test('it should return a declaration version with ID', async () => {
+    const result = await createADeclarationModernSlaveryVersion(context, declarationModernSlavery.id);
+
+    expect(result.id).toBeDefined();
+    expect(typeof result.id).toEqual('string');
+    expect(result.id.length).toBeGreaterThan(0);
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/api/helpers/create-a-declaration-modern-slavery-version/index.test.ts:33]: Good shout - we have this in lots of places. Have created EMS-4036 to tackle in one swoop.

File: src/api/helpers/create-a-declaration-modern-slavery/index.test.ts (hunk 4 above)
Comment (ttbarnes) [reply src/api/helpers/create-a-declaration-modern-slavery/index.test.ts:84]: Have updated.

File: src/api/helpers/create-a-declaration-modern-slavery/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  test('it should return a declaration modern slavery with ID', async () => {
+    const result = await createADeclarationModernSlavery(context, declaration.id);
+
+    expect(result.id).toBeDefined();
+    expect(typeof result.id).toEqual('string');
+    expect(result.id.length).toBeGreaterThan(0);
+  });
+
+  test('it should return a declaration ID', async () => {
```
Comment (Zainzzkk) [inline src/api/helpers/create-a-declaration-modern-slavery/index.test.ts:29]: I just realised.  Should these not be its? and should start with should :P

Comment (ttbarnes) [reply src/api/helpers/create-a-declaration-modern-slavery/index.test.ts:29]: We have over 300 instances of this, I believe from the start of the project. Can tackle this in another PR.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**